### DISCUSSION TITLE
Lesson 4: Issues after installing Nodejs

### DISCUSSION THREAD

**Discussion by itgav** [smartcontractkit/full-blockchain-solidity-course-py#1786] [General]
OS: Windows 11

After installing Nodejs on my system, I was having issues calling Python commands, running deploy.py, some Python libraries weren't showing up as installed. The issue was due to two separate instances of Python on my machine after the Nodejs install. I already had Python installed on my machine and when installing Nodejs another instance of Python was installed with a new directory.

The normal Python installer installs Python to the following directory: "C:\Users\Username\AppData\Local\Programs\Python"
However, Nodejs installs Python to a different directory: "C:\Python"

So, with the new Python path and the installation setting environment variables, when I would call Python in the terminal it was looking at the directory associated with Nodejs Python download. The Nodejs Python folder didn't migrate any of the Python libraries or anything, just installed plain Python.

Ideally, I would've liked to merge my old Python directory into the one created by Nodejs but I couldn't figure out a seamless way to do so in a small amount of time. So, there are two easy solutions with different steps for each. 1) use the original directory of Python; or, 2) use the Nodejs directory. I chose to use the original directory because Nodejs installed Python 3.11 and there were some libraries that weren't updated for 3.11 yet. You can change which version Nodejs installs, but again I was looking for a quick fix.

**Using original Python directory/files**
1.  Remove environment variables for the file path associated with the Python that was downloaded while installing Nodejs (moving them to the bottom of the env variable hierarchy might be sufficient, not sure).
2.  VS Code was still having issues because it was still using the Nodejs Python directory. In VS settings, need to change the Python interpreter to the original Python path.

**Using Python downloaded by Nodejs**
1. Remove environment variables associated with the old Python directory
2. re-install libraries

**Comment by cromewar** [comment]
Thanks for this information @itgav 
Also, in order to avoid problems related to python you can use python virtualenv to create an isolated python container so you can install just the libraries you need for a project.

Finally windows has poor library and dependency management, consider use WSL2 and manage your developer tools there.

At the end, did you find a solution which fulfill your needs?

**Reply by itgav** [reply]
Yes, I just changed the env variables and VS code interpreter. I appreciate the alternatives!

**Reply by cromewar** [reply]
Feel  free to ask any other question or issue you have, I'll do my best to help out.
